ckanext/cnra_schema/helpers.py

```diff
@@ -0,0 +1,75 @@
+from ast import literal_eval
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+def composite_repeating_get_formatted_contact_address_dict(package_dict_field):
+    address_line1 = package_dict_field['address']
+    address_line2 = '{0}, {1} {2}'.format(package_dict_field['city'], package_dict_field['state'],
+                                          package_dict_field['postalCode'])
+    address_line3 = package_dict_field['country']
+
+    address_dict = {
+        'addressLine1': address_line1,
+        'addressLine2': address_line2,
+        'addressLine3': address_line3
+    }
+
+    return address_dict
+
+
+def is_composite_field_populated(package_dict, field):
+    """"
+    This function checks if a composite field populated. Non-composite fields will return True by default.
+    """
+    field_name = field.get('field_name')
+
+    if package_dict.get(field_name) and field_name not in 'spatial_details' \
+            and field.get('preset') in ['composite', 'composite_repeating', 'contact_address_composite_repeating',
+                                        'cnra_composite_repeating']:
```

Comment:
The readability of this if statement can improved by defining the presets list in a variable. Pep8 also suggests using parenthesis for multiline if statements.
```
composite_presets = [
    'composite', 'composite_repeating',
    'contact_address_composite_repeating', 'cnra_composite_repeating'
]
field_name = field.get('field_name')
if (package_dict.get(field_name) 
        and field_name != 'spatial_details' 
        and field.get('preset') in composite_presets):
    subfield_literal_eval = {}
```

Reply:
Another issue is that the if statement is checking if field_name is not in the string `spatial_details`. Either `spatial_details` should be a list or you want to use `!=`. Otherwise potential field names "spatial" and "details" will fail this check.